object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedObjectStore.java

```diff
@@ -238,6 +255,33 @@ private Optional<ContextualConfigObject<T>> processUpsertResult(
     return optionalResult;
   }
 
+  private ContextualConfigObject<T> processDeleteResult(
+      RequestContext requestContext, ContextSpecificConfig deletedConfig) {
+    Optional<ContextualConfigObject<T>> optionalObject =
+        ContextualConfigObjectImpl.tryBuild(deletedConfig, this::buildDataFromValue);
+
+    ContextualConfigObject<T> contextualConfigObject;
+    if (optionalObject.isPresent()) {
+      contextualConfigObject = optionalObject.get();
+      configChangeEventGeneratorOptional.ifPresent(
+          configChangeEventGenerator ->
+              configChangeEventGenerator.sendDeleteNotification(
+                  requestContext,
+                  this.buildClassNameForChangeEvent(contextualConfigObject.getData()),
+                  deletedConfig.getContext(),
+                  this.buildValueForChangeEvent(contextualConfigObject.getData())));
+    } else {
+      contextualConfigObject =
+          new ContextualConfigObjectImpl<>(
+              deletedConfig.getContext(),
+              null,
```

Comment:
I've verified that there are consumers which rely on the data field being non-null. Didn't quite get the part regarding subtype 😅. Could you please elaborate on that?

Reply:
> I've verified that there are consumers which rely on the data field being non-null.

OK so sounds like we need another option. What I meant with a subtype was something like 

``` class DeletedContextualConfigObject extends ContextualConfigObjectImpl {
// Change constructor to not require a config
```

This helps in this spot, but at the end of the day either way - we need to make sure getConfig() returns something for those existing consumers. Maybe that's an empty config? The subclass doesn't really buy us much there though - so if that's what we're doing, I wouldn't bother. the subclass would be useful if we want to put some logic into a getConfig call or, for example, deprecate the returned config on delete only.

Reply:
Hmm, got it. One solution I see is that we add a new abstract method which builds a default instance of the proto message(basically a default/empty config as you had mentioned). But this would involve adding the method in all the places where `IdentifiedObjectStore` is used. What do you think? 🤔

Reply:
Got it. I've tried to make the changes as per 1. But they don't look too clean 😅. What do you think? 🤔